src/repr/src/scalar.rs

```diff
@@ -3728,6 +3732,118 @@ impl Arbitrary for SqlScalarType {
     }
 }
 
+/// The type of a [`Datum`] as it is represented.
+///
+/// Each variant here corresponds to one or more variants of [`SqlScalarType`].
+///
+/// There is a direct correspondence between `Datum` variants and `ScalarType`
+/// variants.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Ord, PartialOrd, Hash, MzReflect)]
+pub enum ReprScalarType {
```

Comment:
>
```
/// There is a direct correspondence between `Datum` variants and `ScalarType`
/// variants.
```

`ReprScalarType` variants?

Reply:
Also, maybe "direct correspondence" could be expanded a bit. E.g., it's not a bijection, because of `Datum::False` and `Datum::True` are both `ReprScalarType::Bool`. Maybe this is the only exception from being a bijection?